src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -51,6 +54,112 @@ public String getColor() {
     }
 
     public List<Section> getSections() {
-        return sections;
+        return sections.list();
+    }
+
+    private Station findFirstUpStation() {
+        return this.sections.findFirstUpStation();
+    }
+
+    public Station findUpStation() {
+        Station downStation = this.findFirstUpStation();
+
+        while (downStation != null) {
+            Station finalDownStation = downStation;
+            Optional<Section> nextLineStation = findLineStation(it -> it.getDownStation() == finalDownStation);
+            if (!nextLineStation.isPresent()) {
+                break;
+            }
+            downStation = nextLineStation.get().getUpStation();
+        }
+
+        return downStation;
+    }
+
+    private boolean matchAnyStation(List<Station> stations, Station station) {
+        return stations.stream().anyMatch(it -> it == station);
+    }
+
+    private boolean matchNoneStation(List<Station> stations, Station station) {
+        return !matchAnyStation(stations, station);
+    }
+
+    private Optional<Section> findLineStation(Predicate<Section> express) {
+        return this.getSections().stream()
+                .filter(express)
+                .findFirst();
+    }
+
+    public void addLineStation(Station upStation, Station downStation, Integer distance) {
+        List<Station> stations = this.getStations();
+        boolean isUpStationExisted = matchAnyStation(stations, upStation);
+        boolean isDownStationExisted = matchAnyStation(stations, downStation);
+
+        if (isUpStationExisted && isDownStationExisted) {
+            throw new RuntimeException("이미 등록된 구간 입니다.");
+        }
+
+        if (!stations.isEmpty() && matchNoneStation(stations, upStation) &&
+                matchNoneStation(stations, downStation)) {
+            throw new RuntimeException("등록할 수 없는 구간 입니다.");
+        }
+
+        if (stations.isEmpty()) {
+            this.sections.add(new Section(this, upStation, downStation, distance));
+            return;
+        }
+
+        if(isUpStationExisted) {
+            findLineStation(it -> it.getUpStation() == upStation)
+                    .ifPresent(it -> it.updateUpStation(downStation, distance));
+            this.sections.add(new Section(this, upStation, downStation, distance));
+            return;
+        }
+
+        findLineStation(it -> it.getDownStation() == downStation)
+            .ifPresent(it -> it.updateDownStation(upStation, distance));
+        this.sections.add(new Section(this, upStation, downStation, distance));
+    }
+
+    public List<Station> getStations() {
+        if (this.getSections().isEmpty()) {
+            return Arrays.asList();
```

Comment:
빈 배열일 경우 Collections의 API를 사용해보아도 좋을 것 같습니다 :)
```suggestion
            return Collections.emptyList();
```

Reply:
넵!